diff --git a/src/freedreno/vulkan/tu_knl_kgsl.cc b/src/freedreno/vulkan/tu_knl_kgsl.cc
index c01160ea..76dce4dd 100644
--- a/src/freedreno/vulkan/tu_knl_kgsl.cc
+++ b/src/freedreno/vulkan/tu_knl_kgsl.cc
@@ -1755,8 +1755,7 @@ VkResult
 tu_knl_kgsl_load(struct tu_instance *instance, int fd)
 {
    if (instance->vk.enabled_extensions.KHR_display) {
-      return vk_errorf(instance, VK_ERROR_INITIALIZATION_FAILED,
-                       "I can't KHR_display");
+      mesa_logw("Ignoring VK_KHR_display on KGSL; no display-device support on this path");
    }

    struct tu_physical_device *device = (struct tu_physical_device *)
